compiler/rustc_privacy/src/lib.rs

```diff
@@ -777,6 +777,13 @@ impl<'tcx> Visitor<'tcx> for EmbargoVisitor<'tcx> {
                         // enum is not, make the enum reachable as well.
                         self.update(item.def_id, variant_level);
                     }
+                    if let Some(hir_id) = variant.data.ctor_hir_id() {
+                        let ctor_def_id = self.tcx.hir().local_def_id(hir_id);
+                        let ctor_level = self.get(ctor_def_id);
+                        if ctor_level.is_some() {
+                            self.reach(item.def_id, ctor_level).ty();
```

Comment:
```suggestion
                            self.update(item.def_id, ctor_level);
```
Slight simplification.
We actually have a nearly identical logic above for reachable variants making their enums reachable.

Reply:
Should this cap the accessibility to reachable? With this implementation the type might be updated with exported level. Not sure how much this matters in practice, but the difference can be observed with `missing_docs` lint for example.

Reply:
Yes, you are right, this use of `update` in variants was a pre-existing issue.
Then we should use `self.reach(item.def_id, ctor_level).ty()` in all these three cases.